src/Cursor.re

```diff
@@ -1,34 +1,46 @@
-type t = {
-  line: int,
-  column: int,
-};
+open EditorCoreTypes;
+
+type t =
+  Location.t = {
+    line: Index.t,
+    column: Index.t,
+  };
 
-let create = (~line, ~column, ()) => {line, column};
+let create = Location.create;
+
+let show = cursor =>
+  Printf.sprintf(
+    "Line %d column %d\n",
+    Index.toOneBased(cursor.line),
+    Index.toZeroBased(cursor.column),
+  );
 
-let show = v => Printf.sprintf("Line %d column %d\n", v.line, v.column);
+let getLine = () => Index.fromOneBased(Native.vimCursorGetLine());
 
-let getLine = Native.vimCursorGetLine;
-let getColumn = Native.vimCursorGetColumn;
+let getColumn = () => Index.fromZeroBased(Native.vimCursorGetColumn());
+let get = () => create(~line=getLine(), ~column=getColumn());
```

Comment:
Also found this function as `_createCursorFromCurrent` in `Vim` and moved it here, but seems like it might as well have used `getLocation`.